## ETL:
### Cargamos el conjunto de datos y visualizamos las primeras filas para entender su estructura.

In [1]:
#Importamos las librerias
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
import warnings
warnings.filterwarnings("ignore")

In [3]:
#Cargamos nuestros datos y visualizamos
datos_pacientes = pd.read_excel(r"C:\Users\aylen\OneDrive\Escritorio\M6\Proyecto Integrador\Propuesta 1\BBDD_Hospitalización.xlsx")
datos_pacientes.head(30)

EDAD DIABETES HOSPITALIZACIÓN ULTIMO MES    PSA BIOPSIAS PREVIAS  \
0   53.0       NO                         NO   4.00               NO   
1   56.0       NO                         NO   7.70               NO   
2   57.0       NO                         NO  29.00               SI   
3   56.0       NO                         NO   7.00               NO   
4   55.0       NO                         NO  29.00               SI   
5   55.0       SI                         SI   4.00               NO   
6   53.0       NO                         NO   7.00               NO   
7   67.0       NO                         NO   6.20               NO   
8   69.0       NO                         NO   6.00               NO   
9   44.0       NO                         NO    NaN               NO   
10  54.0       NO                         NO   7.70               NO   
11  68.0       NO                         NO   6.20               NO   
12  69.0       NO                         NO   5.70               NO   
13  43.0       NO                         NO   6.00               NO   
14  65.0       NO                         NO   4.30               NO   
15  62.0       NO                         NO   7.00               NO   
16  65.0       SI                         NO   5.19               NO   
17  71.0       NO                         NO   4.50               NO   
18  65.0       NO                         NO   4.50               NO   
19  62.0       SI                         NO   7.00               NO   
20  64.0       SI                         NO   5.70               NO   
21  71.0       NO                         NO   4.30               NO   
22  69.0       SI                         NO  14.00               NO   
23  69.0       NO                         NO  14.00               NO   
24  77.0       NO                         NO  10.00               NO   
25  75.0       NO                         NO   9.00               NO   
26  75.0       NO                         NO  25.00               NO   
27  77.0       NO                         NO  91.80               NO   
28  75.0       NO                         NO  21.00               NO   
29  77.0       NO                         NO  98.00               NO   

   VOLUMEN PROSTATICO ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS  \
0                  SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
1                  SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
2                  SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
3                  SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
4                  SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
5                  SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
6                  SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
7                  SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
8                  NO          FLUOROQUINOLONA_AMINOGLICOSIDO   
9                  NO          FLUOROQUINOLONA_AMINOGLICOSIDO   
10                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
11                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
12                 NO          FLUOROQUINOLONA_AMINOGLICOSIDO   
13                 NO          FLUOROQUINOLONA_AMINOGLICOSIDO   
14                 NO          FLUOROQUINOLONA_AMINOGLICOSIDO   
15                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
16                 SI            CEFALOSPORINA_AMINOGLUCOCIDO   
17                 SI                           OROQUINOLONAS   
18                 NO          FLUOROQUINOLONA_AMINOGLICOSIDO   
19                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
20                 SI            CEFALOSPORINA_AMINOGLUCOCIDO   
21                 SI                           OROQUINOLONAS   
22                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
23                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
24                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
25                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
26              

In [4]:
#Estructura del DataFrame
datos_pacientes.shape

(570, 20)

Aqui podemos visualizar que tenemos 570 registros y 20 columnas que representan nuestras variables

In [5]:
#Corroboramos los tipos de variables que tenemos
datos_pacientes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 570 entries, 0 to 569
Data columns (total 20 columns):
 #   Column                                                                     Non-Null Count  Dtype  
---  ------                                                                     --------------  -----  
 0   EDAD                                                                       568 non-null    float64
 1   DIABETES                                                                   568 non-null    object 
 2   HOSPITALIZACIÓN ULTIMO MES                                                 568 non-null    object 
 3   PSA                                                                        564 non-null    float64
 4   BIOPSIAS PREVIAS                                                           566 non-null    object 
 5   VOLUMEN PROSTATICO                                                         567 non-null    object 
 6   ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS                   

In [6]:
datos_pacientes.columns

Index(['EDAD', 'DIABETES', 'HOSPITALIZACIÓN ULTIMO MES', 'PSA',
       'BIOPSIAS PREVIAS', 'VOLUMEN PROSTATICO',
       'ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS', 'NUMERO DE MUESTRAS TOMADAS',
       'CUP', 'ENF. CRONICA PULMONAR OBSTRUCTIVA', 'BIOPSIA',
       'NUMERO DE DIAS POST BIOPSIA EN QUE SE PRESENTA LA COMPLICACIÓN INFECCIOSA',
       'FIEBRE', 'ITU', 'TIPO DE CULTIVO', 'AGENTE AISLADO',
       'PATRON DE RESISTENCIA', 'HOSPITALIZACION', 'DIAS HOSPITALIZACION MQ',
       'DIAS HOSPITALIZACIÓN UPC'],
      dtype='object')

### Verificamos registros duplicados


In [7]:
filas_duplicadas = datos_pacientes[datos_pacientes.duplicated()]
print("Filas duplicadas:")
print(filas_duplicadas)

Filas duplicadas:
     EDAD DIABETES HOSPITALIZACIÓN ULTIMO MES   PSA BIOPSIAS PREVIAS  \
32   73.0       NO                         NO   7.5               NO   
42   62.0       NO                         NO  16.0               NO   
43   69.0       NO                         NO  17.0               SI   
45   66.0       NO                         NO   6.4               NO   
49   68.0       NO                         NO  20.0               NO   
..    ...      ...                        ...   ...              ...   
555  67.0       NO                         NO   8.7               NO   
557  59.0       NO                         NO  16.0               NO   
560  57.0       NO                         NO   4.8               NO   
562  78.0       NO                         NO   9.3               NO   
569   NaN      NaN                        NaN   NaN              NaN   

    VOLUMEN PROSTATICO ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS  \
32                  SI            CEFALOSPORINA_AMI

In [8]:
datos_pacientes_sin_duplicados = datos_pacientes.drop_duplicates().reset_index(drop=True)
datos_pacientes_sin_duplicados

EDAD DIABETES HOSPITALIZACIÓN ULTIMO MES    PSA BIOPSIAS PREVIAS  \
0    53.0       NO                         NO   4.00               NO   
1    56.0       NO                         NO   7.70               NO   
2    57.0       NO                         NO  29.00               SI   
3    56.0       NO                         NO   7.00               NO   
4    55.0       NO                         NO  29.00               SI   
..    ...      ...                        ...    ...              ...   
456  64.0       NO                         NO   4.80               NO   
457  69.0       NO                         NO   4.43              NaN   
458  67.0       NO                         NO   8.00               NO   
459  59.0       NO                         NO   9.00               SI   
460   NaN      NaN                        NaN    NaN              NaN   

    VOLUMEN PROSTATICO ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS  \
0                   SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
1                   SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
2                   SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
3                   SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
4                   SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
..                 ...                                     ...   
456                 SI                                   OTROS   
457                NaN          FLUOROQUINOLONA_AMINOGLICOSIDO   
458                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
459                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
460                NaN                                     NaN   

     NUMERO DE MUESTRAS TOMADAS  CUP ENF. CRONICA PULMONAR OBSTRUCTIVA  \
0                          12.0   NO                                NO   
1                          12.0   NO                                NO   
2                          24.0   NO                                NO   
3                          12.0   NO                                NO   
4                          24.0   NO                                NO   
..                          ...  ...                               ...   
456                        12.0   NO                                NO   
457                        20.0  NaN                                NO   
458                        12.0   NO                               NaN   
459                        12.0   NO                               NaN   
460                         NaN  NaN                               NaN   

                       BIOPSIA  \
0                          NEG   
1                          NEG   
2                          NEG   
3                          NEG   
4                          NEG   
..                         ...   
456  ADENOCARCINOMA GLEASON 7    
457                        NEG   
458  ADENOCARCINOMA GLEASON 7    
459                        NEG   
460                        NaN   

    NUMERO DE DIAS POST BIOPSIA EN QUE SE PRESENTA LA COMPLICACIÓN INFECCIOSA  \
0                                                    1                          
1                                                    1                          
2                                                    1                          
3                                                    1                          
4                                                    1                          
..                                                 ...                          
456                                                 NO                          
457                                                 NO                          
458                                                 NO                          
459                                                 NO                          
460                                                NaN                          

    FIEBRE  ITU TIPO DE CULTIVO AGENTE AISLADO  \
0       SI   N

In [9]:
datos_pacientes_sin_duplicados.shape

(461, 20)

Observamos que despues de eliminar los registros duplicados obtenemos un total de 461 registros

### Verificamos datos faltantes

In [10]:
print(datos_pacientes_sin_duplicados.isnull().sum())

EDAD                                                                          1
DIABETES                                                                      1
HOSPITALIZACIÓN ULTIMO MES                                                    1
PSA                                                                           5
BIOPSIAS PREVIAS                                                              3
VOLUMEN PROSTATICO                                                            2
ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS                                       1
NUMERO DE MUESTRAS TOMADAS                                                    1
CUP                                                                           2
ENF. CRONICA PULMONAR OBSTRUCTIVA                                             3
BIOPSIA                                                                       1
NUMERO DE DIAS POST BIOPSIA EN QUE SE PRESENTA LA COMPLICACIÓN INFECCIOSA     1
FIEBRE                                  

In [11]:
porcentaje_nulos_por_columna = round((datos_pacientes_sin_duplicados.isnull().sum() / datos_pacientes_sin_duplicados.shape[0]) * 100,2)
print("Porcentaje de valores nulos por columna:")
print(porcentaje_nulos_por_columna)

Porcentaje de valores nulos por columna:
EDAD                                                                         0.22
DIABETES                                                                     0.22
HOSPITALIZACIÓN ULTIMO MES                                                   0.22
PSA                                                                          1.08
BIOPSIAS PREVIAS                                                             0.65
VOLUMEN PROSTATICO                                                           0.43
ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS                                      0.22
NUMERO DE MUESTRAS TOMADAS                                                   0.22
CUP                                                                          0.43
ENF. CRONICA PULMONAR OBSTRUCTIVA                                            0.65
BIOPSIA                                                                      0.22
NUMERO DE DIAS POST BIOPSIA EN QUE SE PRESENTA LA COMPLIC

In [12]:
cantidad_registros_nulos = len(datos_pacientes_sin_duplicados[datos_pacientes_sin_duplicados.isna().any(axis=1)])
porcentaje = round((cantidad_registros_nulos / len(datos_pacientes_sin_duplicados)) * 100, 2)
print(f'Hay {cantidad_registros_nulos} registros con al menos un valor nulo, lo que representa el {porcentaje}% del total de registros.')

Hay 28 registros con al menos un valor nulo, lo que representa el 6.07% del total de registros.


In [13]:
nuevo_nombre = "datos_modificados"
datos_modificados = datos_pacientes_sin_duplicados.copy() 
datos_modificados.name = nuevo_nombre

In [14]:
datos_modificados =datos_modificados.dropna()
datos_modificados

EDAD DIABETES HOSPITALIZACIÓN ULTIMO MES   PSA BIOPSIAS PREVIAS  \
0    53.0       NO                         NO   4.0               NO   
1    56.0       NO                         NO   7.7               NO   
2    57.0       NO                         NO  29.0               SI   
3    56.0       NO                         NO   7.0               NO   
4    55.0       NO                         NO  29.0               SI   
..    ...      ...                        ...   ...              ...   
452  55.0       NO                         NO   5.0               NO   
453  70.0       SI                         NO  18.0               NO   
454  75.0       NO                         NO  75.0               NO   
455  67.0       NO                         NO   6.0               NO   
456  64.0       NO                         NO   4.8               NO   

    VOLUMEN PROSTATICO ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS  \
0                   SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
1                   SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
2                   SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
3                   SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
4                   SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
..                 ...                                     ...   
452                 NO          FLUOROQUINOLONA_AMINOGLICOSIDO   
453                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
454                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
455                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
456                 SI                                   OTROS   

     NUMERO DE MUESTRAS TOMADAS CUP ENF. CRONICA PULMONAR OBSTRUCTIVA  \
0                          12.0  NO                                NO   
1                          12.0  NO                                NO   
2                          24.0  NO                                NO   
3                          12.0  NO                                NO   
4                          24.0  NO                                NO   
..                          ...  ..                               ...   
452                        12.0  NO                                NO   
453                        12.0  NO                                NO   
454                        12.0  NO                                NO   
455                        12.0  NO                                NO   
456                        12.0  NO                                NO   

                       BIOPSIA  \
0                          NEG   
1                          NEG   
2                          NEG   
3                          NEG   
4                          NEG   
..                         ...   
452  ADENOCARCINOMA GLEASON 7    
453                        NEG   
454  ADENOCARCINOMA GLEASON 8    
455   ADENOCARCINOMA GLEASON 6   
456  ADENOCARCINOMA GLEASON 7    

    NUMERO DE DIAS POST BIOPSIA EN QUE SE PRESENTA LA COMPLICACIÓN INFECCIOSA  \
0                                                    1                          
1                                                    1                          
2                                                    1                          
3                                                    1                          
4                                                    1                          
..                                                 ...                          
452                                                 NO                          
453                                                 NO                          
454                                                 NO                          
455                                                 NO                          
456                                                 NO                          

    FIEBRE ITU TIPO DE CULTIVO AGENTE AISLADO  \
0       SI  NO              NO         

In [15]:
print(datos_modificados.isnull().sum())

EDAD                                                                         0
DIABETES                                                                     0
HOSPITALIZACIÓN ULTIMO MES                                                   0
PSA                                                                          0
BIOPSIAS PREVIAS                                                             0
VOLUMEN PROSTATICO                                                           0
ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS                                      0
NUMERO DE MUESTRAS TOMADAS                                                   0
CUP                                                                          0
ENF. CRONICA PULMONAR OBSTRUCTIVA                                            0
BIOPSIA                                                                      0
NUMERO DE DIAS POST BIOPSIA EN QUE SE PRESENTA LA COMPLICACIÓN INFECCIOSA    0
FIEBRE                                              

In [16]:
datos_modificados.reset_index(inplace=True, drop=True)
datos_modificados

EDAD DIABETES HOSPITALIZACIÓN ULTIMO MES   PSA BIOPSIAS PREVIAS  \
0    53.0       NO                         NO   4.0               NO   
1    56.0       NO                         NO   7.7               NO   
2    57.0       NO                         NO  29.0               SI   
3    56.0       NO                         NO   7.0               NO   
4    55.0       NO                         NO  29.0               SI   
..    ...      ...                        ...   ...              ...   
428  55.0       NO                         NO   5.0               NO   
429  70.0       SI                         NO  18.0               NO   
430  75.0       NO                         NO  75.0               NO   
431  67.0       NO                         NO   6.0               NO   
432  64.0       NO                         NO   4.8               NO   

    VOLUMEN PROSTATICO ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS  \
0                   SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
1                   SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
2                   SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
3                   SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
4                   SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
..                 ...                                     ...   
428                 NO          FLUOROQUINOLONA_AMINOGLICOSIDO   
429                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
430                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
431                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
432                 SI                                   OTROS   

     NUMERO DE MUESTRAS TOMADAS CUP ENF. CRONICA PULMONAR OBSTRUCTIVA  \
0                          12.0  NO                                NO   
1                          12.0  NO                                NO   
2                          24.0  NO                                NO   
3                          12.0  NO                                NO   
4                          24.0  NO                                NO   
..                          ...  ..                               ...   
428                        12.0  NO                                NO   
429                        12.0  NO                                NO   
430                        12.0  NO                                NO   
431                        12.0  NO                                NO   
432                        12.0  NO                                NO   

                       BIOPSIA  \
0                          NEG   
1                          NEG   
2                          NEG   
3                          NEG   
4                          NEG   
..                         ...   
428  ADENOCARCINOMA GLEASON 7    
429                        NEG   
430  ADENOCARCINOMA GLEASON 8    
431   ADENOCARCINOMA GLEASON 6   
432  ADENOCARCINOMA GLEASON 7    

    NUMERO DE DIAS POST BIOPSIA EN QUE SE PRESENTA LA COMPLICACIÓN INFECCIOSA  \
0                                                    1                          
1                                                    1                          
2                                                    1                          
3                                                    1                          
4                                                    1                          
..                                                 ...                          
428                                                 NO                          
429                                                 NO                          
430                                                 NO                          
431                                                 NO                          
432                                                 NO                          

    FIEBRE ITU TIPO DE CULTIVO AGENTE AISLADO  \
0       SI  NO              NO         

### Verificamos los tipos de datos

In [17]:
datos_modificados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 433 entries, 0 to 432
Data columns (total 20 columns):
 #   Column                                                                     Non-Null Count  Dtype  
---  ------                                                                     --------------  -----  
 0   EDAD                                                                       433 non-null    float64
 1   DIABETES                                                                   433 non-null    object 
 2   HOSPITALIZACIÓN ULTIMO MES                                                 433 non-null    object 
 3   PSA                                                                        433 non-null    float64
 4   BIOPSIAS PREVIAS                                                           433 non-null    object 
 5   VOLUMEN PROSTATICO                                                         433 non-null    object 
 6   ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS                   

In [18]:
datos_modificados["EDAD"]= datos_modificados["EDAD"].astype(int)
datos_modificados["DIAS HOSPITALIZACION MQ"]= datos_modificados["DIAS HOSPITALIZACION MQ"].astype(int)
datos_modificados["NUMERO DE MUESTRAS TOMADAS"]= datos_modificados["NUMERO DE MUESTRAS TOMADAS"].astype(int)
datos_modificados["DIAS HOSPITALIZACIÓN UPC"]= datos_modificados["DIAS HOSPITALIZACIÓN UPC"].astype(int)

In [19]:
datos_modificados["NUMERO DE DIAS POST BIOPSIA EN QUE SE PRESENTA LA COMPLICACIÓN INFECCIOSA"] = datos_modificados["NUMERO DE DIAS POST BIOPSIA EN QUE SE PRESENTA LA COMPLICACIÓN INFECCIOSA"].replace("NO", 0)

In [20]:
datos_modificados[datos_modificados["NUMERO DE DIAS POST BIOPSIA EN QUE SE PRESENTA LA COMPLICACIÓN INFECCIOSA"]== 0]

EDAD DIABETES HOSPITALIZACIÓN ULTIMO MES   PSA BIOPSIAS PREVIAS  \
25     77       NO                         NO  91.8               NO   
26     75       NO                         NO  21.0               NO   
27     77       NO                         NO  98.0               NO   
28     73       NO                         NO   7.5               NO   
29     63       NO                         NO  10.5               NO   
..    ...      ...                        ...   ...              ...   
428    55       NO                         NO   5.0               NO   
429    70       SI                         NO  18.0               NO   
430    75       NO                         NO  75.0               NO   
431    67       NO                         NO   6.0               NO   
432    64       NO                         NO   4.8               NO   

    VOLUMEN PROSTATICO ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS  \
25                  SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
26                  SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
27                  SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
28                  SI            CEFALOSPORINA_AMINOGLUCOCIDO   
29                  NO          FLUOROQUINOLONA_AMINOGLICOSIDO   
..                 ...                                     ...   
428                 NO          FLUOROQUINOLONA_AMINOGLICOSIDO   
429                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
430                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
431                 SI          FLUOROQUINOLONA_AMINOGLICOSIDO   
432                 SI                                   OTROS   

     NUMERO DE MUESTRAS TOMADAS CUP ENF. CRONICA PULMONAR OBSTRUCTIVA  \
25                            6  NO                          SI, EPOC   
26                           12  NO                          SI, EPOC   
27                            6  NO                          SI, EPOC   
28                           12  NO                          SI, ASMA   
29                           12  NO                          SI, ASMA   
..                          ...  ..                               ...   
428                          12  NO                                NO   
429                          12  NO                                NO   
430                          12  NO                                NO   
431                          12  NO                                NO   
432                          12  NO                                NO   

                        BIOPSIA  \
25   ADENOCARCINOMA GLEASON 10    
26    ADENOCARCINOMA GLEASON 7    
27   ADENOCARCINOMA GLEASON 10    
28                          NEG   
29    ADENOCARCINOMA GLEASON 7    
..                          ...   
428   ADENOCARCINOMA GLEASON 7    
429                         NEG   
430   ADENOCARCINOMA GLEASON 8    
431    ADENOCARCINOMA GLEASON 6   
432   ADENOCARCINOMA GLEASON 7    

     NUMERO DE DIAS POST BIOPSIA EN QUE SE PRESENTA LA COMPLICACIÓN INFECCIOSA  \
25                                                   0                           
26                                                   0                           
27                                                   0                           
28                                                   0                           
29                                                   0                           
..                                                 ...                           
428                                                  0                           
429                                                  0                           
430                                                  0                           
431                                                  0                           
432                                                  0                           

    FIEBRE ITU TIPO DE CULTIVO AGENTE AISLADO PATRON DE RESISTEN

In [21]:
datos_modificados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 433 entries, 0 to 432
Data columns (total 20 columns):
 #   Column                                                                     Non-Null Count  Dtype  
---  ------                                                                     --------------  -----  
 0   EDAD                                                                       433 non-null    int32  
 1   DIABETES                                                                   433 non-null    object 
 2   HOSPITALIZACIÓN ULTIMO MES                                                 433 non-null    object 
 3   PSA                                                                        433 non-null    float64
 4   BIOPSIAS PREVIAS                                                           433 non-null    object 
 5   VOLUMEN PROSTATICO                                                         433 non-null    object 
 6   ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS                   

In [22]:
datos_modificados["HOSPITALIZACIÓN ULTIMO MES"].unique()

array(['NO', 'SI'], dtype=object)

In [23]:
datos_modificados["BIOPSIAS PREVIAS"].unique()

array(['NO', 'SI'], dtype=object)

In [24]:
datos_modificados["VOLUMEN PROSTATICO"].unique()

array(['SI', 'NO'], dtype=object)

In [25]:
datos_modificados["ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS"].unique()

array(['FLUOROQUINOLONA_AMINOGLICOSIDO', 'CEFALOSPORINA_AMINOGLUCOCIDO',
       'OROQUINOLONAS', 'OTROS', 'FLUOROQUINOLONA_AMINOGLICÓSIDO'],
      dtype=object)

Podemos observar que tenemos dos nombres iguales en los que solo varian los tildes por lo que procedemos a unificar

In [26]:
datos_modificados["ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS"]= datos_modificados["ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS"].replace('FLUOROQUINOLONA_AMINOGLICÓSIDO','FLUOROQUINOLONA_AMINOGLICOSIDO')

In [27]:
datos_modificados["CUP"].unique()

array(['NO', 'SI'], dtype=object)

In [28]:
datos_modificados["ENF. CRONICA PULMONAR OBSTRUCTIVA"].unique()

array(['NO', 'SI, EPOC', 'SI, ASMA', 'SI'], dtype=object)

In [29]:
datos_modificados["ENF. CRONICA PULMONAR OBSTRUCTIVA"]= datos_modificados["ENF. CRONICA PULMONAR OBSTRUCTIVA"].replace('SI, ASMA','SI')
datos_modificados["ENF. CRONICA PULMONAR OBSTRUCTIVA"]= datos_modificados["ENF. CRONICA PULMONAR OBSTRUCTIVA"].replace('SI, EPOC','SI')

In [30]:
datos_modificados["BIOPSIA"].unique()

array(['NEG', 'ADENOCARCINOMA GLEASON 6 ', 'ADENOCARCINOMA GLEASON 7 ',
       'ADENOCARCINOMA GLEASON 6', 'ADENOCARCINOMA GLEASON 10 ',
       'ADENOCARCINOMA GLEASON 9 ', 'ADENOCARCINOMA GLEASON 8 ',
       'PROSTATITIS', 'ADENOCARCINOMA GLEASON 7',
       'HIPERPLASIA PROSTATICA',
       'CARCINOMA INDIFERENCIADO DE CELULAS CLARAS'], dtype=object)

In [31]:
datos_modificados["BIOPSIA"] = datos_modificados["BIOPSIA"].str.rstrip()

In [32]:
datos_modificados["FIEBRE"].unique()

array(['SI', 'NO'], dtype=object)

In [33]:
datos_modificados["ITU"].unique()

array(['NO', 'SI'], dtype=object)

In [34]:
datos_modificados["TIPO DE CULTIVO"].unique()

array(['NO', 'HEMOCULTIVO', 'UROCULTIVO', 'HEMOCULTIVO Y UROCULTIVO'],
      dtype=object)

In [35]:
datos_modificados["AGENTE AISLADO"].unique()

array(['NO', 'E.COLI', 'PSEUDOMONAS AERUGINOSA'], dtype=object)

In [36]:
datos_modificados["PATRON DE RESISTENCIA"].unique()

array(['NO', ' AMPI R, CIPRO R, GENTA R, SULFA M R',
       'RESISTENTE A AMPI, CIPRO Y GENTA',
       'RESISTENTE A AMPI, SULFA, CEFADROXILO, CEFUROXIMO, CIPRO Y CEFEPIME, CEFOTAXIMA',
       'MULTI SENSIBLE'], dtype=object)

In [37]:
datos_modificados["PATRON DE RESISTENCIA"]= datos_modificados["PATRON DE RESISTENCIA"].replace(' AMPI R, CIPRO R, GENTA R, SULFA M R','SI')
datos_modificados["PATRON DE RESISTENCIA"]= datos_modificados["PATRON DE RESISTENCIA"].replace('RESISTENTE A AMPI, CIPRO Y GENTA','SI')
datos_modificados["PATRON DE RESISTENCIA"]= datos_modificados["PATRON DE RESISTENCIA"].replace('RESISTENTE A AMPI, SULFA, CEFADROXILO, CEFUROXIMO, CIPRO Y CEFEPIME, CEFOTAXIMA','SI')
datos_modificados["PATRON DE RESISTENCIA"]= datos_modificados["PATRON DE RESISTENCIA"].replace('MULTI SENSIBLE','SI')

In [38]:
datos_modificados = datos_modificados.rename(columns={'EDAD': 'Edad','DIABETES': 'Diabetes','HOSPITALIZACIÓN ULTIMO MES': 'Hosp_ultimo_mes', 'PSA': 'PSA',
                                                  'BIOPSIAS PREVIAS': 'Biopsias_previas','VOLUMEN PROSTATICO': 'Vol_prostatico',
                                                  'ANTIBIOTICO UTILIAZADO EN LA PROFILAXIS': 'Antibiotico_profilaxis',
                                                  'NUMERO DE MUESTRAS TOMADAS': 'Num_muestras','CUP': 'CUP',
                                                  'ENF. CRONICA PULMONAR OBSTRUCTIVA': 'Enf_cronica_pulm_obstructiva',
                                                  'BIOPSIA': 'Biopsia',
                                                  'NUMERO DE DIAS POST BIOPSIA EN QUE SE PRESENTA LA COMPLICACIÓN INFECCIOSA': 'Dias_post_biopsia_infeccion',
                                                   'FIEBRE': 'Fiebre','ITU': 'ITU',
                                                   'TIPO DE CULTIVO': 'Tipo_cultivo',
                                                   'AGENTE AISLADO': 'Agente_aislado',
                                                   'PATRON DE RESISTENCIA': 'Patron_resistencia','HOSPITALIZACION': 'Hospitalizacion',
                                                   'DIAS HOSPITALIZACION MQ': 'Dias_hosp_MQ',
                                                   'DIAS HOSPITALIZACIÓN UPC': 'Dias_hosp_UPC'
                                                  })

In [39]:
datos_modificados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 433 entries, 0 to 432
Data columns (total 20 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Edad                          433 non-null    int32  
 1   Diabetes                      433 non-null    object 
 2   Hosp_ultimo_mes               433 non-null    object 
 3   PSA                           433 non-null    float64
 4   Biopsias_previas              433 non-null    object 
 5   Vol_prostatico                433 non-null    object 
 6   Antibiotico_profilaxis        433 non-null    object 
 7   Num_muestras                  433 non-null    int32  
 8   CUP                           433 non-null    object 
 9   Enf_cronica_pulm_obstructiva  433 non-null    object 
 10  Biopsia                       433 non-null    object 
 11  Dias_post_biopsia_infeccion   433 non-null    int64  
 12  Fiebre                        433 non-null    object 
 13  ITU  

In [40]:
datos_limpios = 'BBDD_hospitalizacion_limpio.csv'
datos_modificados.to_csv(datos_limpios, index=False, encoding='utf-8')
print(f'Se guardó el archivo {datos_limpios}')

Se guardó el archivo BBDD_hospitalizacion_limpio.csv
